### PR TITLE
[ROAD-811] feat: added analysisContext to analysis requests

### PR DESCRIPTION
**Pull request by aldanchenko**
### Description

Added analysisContext information (flow and client's organization name) to analysis request.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [x] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: Snyk.Code.Library/Api/ISnykCodeClient.cs [hunk 1]
```diff
@@ -89,5 +89,18 @@ public interface ISnykCodeClient
...
+        /// For better annotation and tracking it contains analysisContext information.
+        /// analysisContext: {
+        ///     flow: ‘specific IDE’,
+        ///     initiator: ‘IDE’, this is important, don’t change it
+        ///     orgDisplayName: ‘client’s snyk organization name’
+        /// }.
+        /// </summary>
+        /// <param name="bundleId">Bundle Id value.</param>
+        /// <returns>Json payload for analysis result request.</returns>
+        string GetAnalysisResultRequestPayload(string bundleId);
```
[the remark this answers is not in the document]
Comment (aldanchenko) [reply Snyk.Code.Library/Api/ISnykCodeClient.cs:104]: What are the options here?

[the remark this answers is not in the document]
Comment (aldanchenko) [reply Snyk.Code.Library/Api/ISnykCodeClient.cs:104]: What are the options?

File: Snyk.Code.Library/Api/SnykCodeClient.cs [hunk 2]
```diff
@@ -259,5 +260,22 @@ public async Task<FiltersDto> GetFiltersAsync()
                 }
             }
         }
+
+        /// <inheritdoc/>
```
[the remark this answers is not in the document]
Comment (aldanchenko) [reply Snyk.Code.Library/Api/SnykCodeClient.cs:264]: Maybe I don't understand it. But we have this - https://github.com/snyk/snyk-visual-studio-plugin/blob/4da6554ca85a01d952c7e662a4e53932e5bb8ac9/Snyk.Code.Library/Api/ISnykCodeClient.cs#L104

File: Snyk.Code.Library/Api/ISnykCodeClient.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (asaf92) [reply Snyk.Code.Library/Api/ISnykCodeClient.cs:104]: Ideally, we'd mock the `HttpClient` and read the payload it generates, but don't worry about it now

**Review by asaf92 — APPROVED** [review]
LGTM!

**Pull request merged**